linfa-svm/src/solver_smo.rs

```diff
@@ -841,14 +863,27 @@ impl<'a, F: Float, K: 'a + Permutable<F>> SolverState<'a, F, K> {
             None
         };
 
+        // Select only vectors that have non-zero influence on the outcome
+        // of future predictions
+        let support_vectors = self.dataset.select(
+            Axis(0),
+            &alpha
+                .iter()
+                .enumerate()
+                .filter(|(_, a)| !a.is_zero())
```

Comment:
can you use a threshold here, because sometimes the solution can be very small but non-zero. I used `1e-5` previously but perhaps a multiple of the `F::eps` would be more sensible

Reply:
I set it to `100 * F::eps` so to get ~1e-5 for f32 and ~2e-14 for f64, just FYI